custom-elements/custom-elements-registry/define.html

```diff
@@ -0,0 +1,178 @@
+<link rel="help" href="https://html.spec.whatwg.org/multipage/scripting.html#element-definition">
+<meta name="author" title="Koji Ishii" href="mailto:kojiishi@gmail.com">
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<body>
+<div id="log"></div>
+<script>
+'use strict';
+(function () {
+  var testable = false;
+  test(function () {
+    assert_true('customElements' in window, '"window.customElements" exists');
+    assert_true('define' in window.customElements, '"window.customElements.define" exists');
+    testable = true;
+  }, '"window.customElements.define" exists');
+  if (!testable)
+    return;
+
+  // 1. If IsConstructor(constructor) is false,
+  // then throw a TypeError and abort these steps.
+  test(function () {
+    assert_throws(new TypeError, function () {
+      window.customElements.define();
+    });
+  }, 'If no arguments, throw a TypeError');
+  test(function () {
+    assert_throws(new TypeError, function () {
+      window.customElements.define("test-define-one-arg");
+    });
+  }, 'If one argument, throw a TypeError');
+  [
+    [ "undefined", undefined ],
+    [ "null", null ],
+    [ "object", {} ],
+    [ "string", "string" ],
+    [ "function", function () {} ],
+  ].forEach(function (t) {
+    test(function () {
+      assert_throws(new TypeError, function () {
+        window.customElements.define("test-define-ctor-" + t[0], t[1]);
+      });
+    }, 'If constructor is ' + t[0] + ', throw a TypeError');
+  });
+
+  // 2. If name is not a valid custom element name,
+  // then throw a SyntaxError and abort these steps.
+  var validCustomElementNames = [
+    // [a-z] (PCENChar)* '-' (PCENChar)*
+    // https://html.spec.whatwg.org/multipage/scripting.html#valid-custom-element-name
+    "a-",
+    "a-a",
+    "aa-",
+    "aa-a",
+    "a-.-_",
+    "a-0123456789",
+    "a-\u6F22\u5B57", // Two CJK Unified Ideographs
+    "a-\uD840\uDC0B", // Surrogate pair U+2000B
+  ];
+  var invalidCustomElementNames = [
+    undefined,
+    null,
+    "",
+    "a",
+    "A",
+    "A-",
+    "0-",
+    "a-A",
+    "a-Z",
+    // name must not be any of the hyphen-containing element names.
+    "annotation-xml",
+    "color-profile",
+    "font-face",
+    "font-face-src",
+    "font-face-uri",
+    "font-face-format",
+    "font-face-name",
+    "missing-glyph",
+  ];
+  validCustomElementNames.forEach(function (name) {
+    test(function () {
+      window.customElements.define(name, class {});
+    }, 'If name is ' + name + ', success');
+  });
+  invalidCustomElementNames.forEach(function (name) {
+    test(function () {
+      assert_throws(new SyntaxError, function () {
+        window.customElements.define(name, class {});
+      });
+    }, 'If name is ' + name + ', throw a SyntaxError');
+  });
+
+  // 3. If this CustomElementsRegistry contains an entry with name name,
+  // then throw a NotSupportedError and abort these steps.
+  test(function () {
```

Comment:
By making them as one test(), it only finds the first failure. You fix it and run again to find the next failure. Isn't it better to find all failures at once?